dali/kernels/signal/resampling_gpu.cuh

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_SIGNAL_RESAMPLING_GPU_CUH_
+#define DALI_KERNELS_SIGNAL_RESAMPLING_GPU_CUH_
+
+#include <cuda_runtime.h>
+#include "dali/kernels/signal/resampling.h"
+#include "dali/core/util.h"
+
+#define SHM_NCHANNELS 16
```

Comment:
```suggestion
constexpr int SHM_NCHANNELS = 16;
```
(possibly with different capitalization)?

Reply:
this gives me this error:
```error: identifier "dali::kernels::signal::resampling::kShmChannels" is undefined in device code```

Reply:
The problem lies with `cuda_min`, which takes a reference. A workaround would be:
`enum : int { kShmChannels = 16 };` - I'm not pushing it, just posting it here for completeness.